app/src/main/java/com/release/keyneez/presentation/main/search/SearchActivity.kt

```diff
@@ -0,0 +1,47 @@
+package com.release.keyneez.presentation.main.search
+
+import android.os.Bundle
+import android.view.KeyEvent
+import androidx.activity.viewModels
+import com.release.keyneez.R
+import com.release.keyneez.databinding.ActivitySearchBinding
+import com.release.keyneez.util.binding.BindingActivity
+import com.release.keyneez.util.extension.hideKeyboard
+import com.release.keyneez.util.extension.setOnSingleClickListener
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class SearchActivity : BindingActivity<ActivitySearchBinding>(R.layout.activity_search) {
+    lateinit var searchAdapter: SearchAdapter
+    private val viewModel by viewModels<SearchViewModel>()
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding.vm = viewModel
+        initSearchBtnClickListener()
+        initHideKeyboard()
+        initSearchBackBtnClickListener()
+    }
+
+    private fun initSearchBtnClickListener() {
+        binding.btnSearchResult.setOnKeyListener { v, keyCode, event ->
+            if ((event.action == KeyEvent.ACTION_DOWN) && (keyCode == KeyEvent.KEYCODE_ENTER)) {
+                // 뷰모델 통신하는 코드 작성하기
+                true
+            } else {
+                false
+            }
+        }
+    }
```

Comment:
```suggestion
    private fun initSearchBtnKeyListener() {
        binding.btnSearchResult.setOnKeyListener { v, keyCode, event ->
            if ((event.action == KeyEvent.ACTION_DOWN) && (keyCode == KeyEvent.KEYCODE_ENTER)) {
                // 뷰모델 통신하는 코드 작성하기
                true
                return@initSearchBtnKeyListener
            }
            false
        }
    }
```
더 섬세한 예외 처리와 가독성 개선을 위해 `else` 사용은 지양해주세요!
한번 early return 패턴에 대해 찾아보셔도 좋을 것 같습니다 :)

Reply:
그리고 이 함수는 클릭 리스너가 아닌 키 리스너를 설정하는 함수이므로 역할에 맞는 네이밍을 하는 것이 좋을 것 같습니다!

Reply:
엇 그 저희 컨벤션에 없는데 어떤 이름으로 하는 것이 좋을까요?

Reply:
![image](https://github.com/Keyneez/Keyneez-AOS-Release/assets/70993562/5f3a05c9-f3fd-4555-b59e-1337140a049e)

클릭 리스너는 초기화 함수 예시 중 하나라고 생각하시면 될 것 같아요! 키 리스너로 네이밍하는 게 좋을 것 같슴다 :)